File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/PathRepository.java
package com.raxim.myscoutee.profile.repository.mongo;

import com.raxim.myscoutee.profile.data.document.mongo.Path;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

import java.util.UUID;

@RepositoryRestResource(collectionResourceRel = "paths", path = "paths")
interface PathRepository extends MongoRepository<Path, UUID>, QuerydslPredicateExecutor<Path> {

}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/PathRepository.kt
package com.raxim.myscoutee.profile.repository.mongo

import com.raxim.myscoutee.profile.data.document.mongo.Path
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import java.util.*

@RepositoryRestResource(
    collectionResourceRel = "paths",
    path = "paths"
)
interface PathRepository : MongoRepository<Path, UUID>,
    QuerydslPredicateExecutor<Path>